### PR TITLE
*: add service subcommand

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5836.

### What is changed and how does it work?
This PR has a similar functionality to #5964 but uses the independent config for starting the TSO server. Besides, we can use the same function to init log for all binaries now. 
Update: 
Replace the original implementation with the cobra.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

File: cmd/pd-server/main.go
Comment (binshi-bing) [inline cmd/pd-server/main.go:46]: nit: can we combine these two parameters into 'service-mode-tso'? This is the subcommand format that I know. If "service tso-service" is TiDB convention, please ignore my comment.

File: cmd/pd-server/main.go
Comment (binshi-bing) [inline cmd/pd-server/main.go:55]: shouldn't it be case insensitive?

File: cmd/pd-server/main.go
Comment (binshi-bing) [inline cmd/pd-server/main.go:60]: After log.Fatal, will the command output be friendly to user like displaying the help info and the supported service modes? If not, probably we want to log error and print help info here.

**Review by binshi-bing — APPROVED** [review]
Left few comments, could you please take a look? The other parts LGTM

**Comment by rleungx** [comment]
Since the PR has been rewritten, need two more LGTM before merging.

**Review by lhy1024 — APPROVED** [review]

File: cmd/pd-server/main.go
Comment (binshi-bing) [inline cmd/pd-server/main.go:98]: so this is why we removed the return data types and have to duplicate server run() and exit code. LGTM

**Review by binshi-bing — APPROVED** [review]
Looks like with Cobra, we have to duplicate server.run and exit code. LGTM

**Comment by lhy1024** [comment]
> Looks like with Cobra, we have to duplicate server.run and exit code. LGTM

I think we can will move some functions to `util.go` after ensuring `todo` in tso server

**Comment by rleungx** [comment]
@JmPotato @nolouch Please take a look when you are available.

**Review by JmPotato — APPROVED** [review]

**Pull request merged**
